src/config_web.py

```diff
@@ -90,84 +90,112 @@
-QUERY_KWARGS['*']['_source']['strict'] = False
-QUERY_KWARGS['GET']['q']['translations'] = DATASOURCE_TRANSLATION_TYPEDEF
-QUERY_KWARGS['POST']['scopes']['translations'] = TRIMMED_DATASOURCE_TRANSLATION_TYPEDEF
-QUERY_KWARGS['GET']['include_tax_tree'] = {'type': bool, 'default': False}
-QUERY_KWARGS['POST']['scopes']['default'] = ["_id", "entrezgene", "ensembl.gene", "retired"]
-QUERY_KWARGS['POST']['q']['jsoninput'] = True
+QUERY_KWARGS["*"].update(SPECIES_TYPEDEF)
+QUERY_KWARGS["*"].update(FIELD_FILTERS)
+QUERY_KWARGS["*"]["_source"]["default"] = DEFAULT_FIELDS
+QUERY_KWARGS["*"]["_source"]["strict"] = False
+QUERY_KWARGS["GET"]["q"]["translations"] = DATASOURCE_TRANSLATION_TYPEDEF
+QUERY_KWARGS["POST"]["scopes"]["translations"] = TRIMMED_DATASOURCE_TRANSLATION_TYPEDEF
+QUERY_KWARGS["GET"]["include_tax_tree"] = {"type": bool, "default": False}
+QUERY_KWARGS["POST"]["scopes"]["default"] = [
+    "_id",
+    "entrezgene",
+    "ensembl.gene",
+    "retired",
+]
+QUERY_KWARGS["POST"]["q"]["jsoninput"] = True
 
 
 # *****************************************************************************
 # Elasticsearch Query Pipeline
 # *****************************************************************************
 ES_QUERY_BUILDER = "web.pipeline.MygeneQueryBuilder"
-AVAILABLE_FIELDS_EXCLUDED = ['all', 'accession_agg', 'refseq_agg']
+AVAILABLE_FIELDS_EXCLUDED = ["all", "accession_agg", "refseq_agg"]
 
 # *****************************************************************************
 # Endpoints Specifics & Others
 # *****************************************************************************
 
 # kwargs for status check
-STATUS_CHECK = {
-    'id': '1017',
-    'index': 'mygene_current'
-}
+STATUS_CHECK = {"id": "1017", "index": "mygene_current"}
 
-# This essentially bypasses the es.get fallback as in myvariant...
+# This essentially bypasses the es.get fallback as in myvariant ...
 # The first regex matched integers, in which case the query becomes against
 # entrezgeneall annotation queries are now multimatch against the following fields
-ANNOTATION_ID_REGEX_LIST = [(re.compile(r'^\d+$'), ['entrezgene', 'retired'])]
+base_regex_pattern = re.compile(r"^\d+$")
+base_field_scope = ["entrezgene", "retired"]
+
+# CURIE ID support
+BIOLINK_MODEL_PREFIX_BIOTHINGS_GENE_MAPPING = {
+    "NCBIGene": {"type": "gene", "field": "entrezgene"},
+    "ENSEMBL": {"type": "gene", "field": "ensembl.gene"},
+    "UniProtKB": {"type": "gene", "field": "uniprot.Swiss-Prot"},
+}
+gene_pattern_field_mapping = []
+for (
+    biolink_model,
+    biothings_parameters,
+) in BIOLINK_MODEL_PREFIX_BIOTHINGS_GENE_MAPPING.items():
+    expression = re.compile(rf"({biolink_model}):(?P<term>[^:]+)", re.I)
+    field_match = biothings_parameters["field"]
+    pattern = (expression, field_match)
+    gene_pattern_field_mapping.append(pattern)
+
+ANNOTATION_ID_REGEX_LIST = [
+    *gene_pattern_field_mapping,
+    (base_regex_pattern, base_field_scope),
```

Comment:
wonder if we can just put two values here instead of using two base variables.

Alternatively, similar to `gene_pattern_field_mapping` list, you can create a `base_regex_list` variable.